src/components/YearSelection.js

```diff
@@ -0,0 +1,64 @@
+import React, { Component } from 'react';
+import {
+  Dropdown,
+  DropdownToggle,
+  DropdownMenu,
+  DropdownItem
+} from 'reactstrap';
+
+const DropdownMenuStyle = {
+  height: '200px',
+  overflow: 'auto'
+};
+
+class YearSelection extends Component {
+  constructor(props) {
+    super(props);
+
+    this.toggle = this.toggle.bind(this);
+    this.state = {
+      dropdownOpen: false,
+      dropDownValue: 'Select a Year'
+    };
+  }
+
+  toggle() {
+    this.setState(prevState => ({
+      dropdownOpen: !prevState.dropdownOpen
+    }));
+  }
+
+  changeValue(e) {
+    this.setState({ dropDownValue: e.currentTarget.textContent });
+  }
+
+  render() {
+    let allYears = [];
+    for (let i = 1990; i < 2019; i++) {
```

Comment:
Can this loop be outside of `render` method, so that the component doesn't loop it every time it needs re-rendering?

Reply:
Moved it outside render